### ISSUE TITLE
Writing to the collection (col.write(modify=true)) breaks the collection

### ISSUE THREAD

**Issue by exc4l** [klieret/AnkiPandas#50]
MWE:
```python
col = Collection(user='testing')
notes = col.notes
selection = notes[notes.has_tag("leech")]
#selection = selection.add_tag(["marked"])
col.notes.update(selection)
col.write(modify=True)
```
This doesn't result in an immediate error but after opening anki the card browser is unable to find notes/cards by tag.

In the testing profile, I do have 32 notes tagged with 'leech'.
After writing when searching by the leech tag it shows no notes.
The notes are still there and if I search for them manually I can see that they indeed do have the leech tag.
If I, after now manually selecting them, do a search again for the leech tag, they do appear but also only the ones I selected manually.

When using the .add_tag() function the new tags do show up in the notes but it's the same with the leech tag. The card browser is only able to find the new tag after I manually selected the cards.

This isn't fixed by anki's "check database" function.

As mentioned before I'm using an older anki version. If you aren't able to reproduce this let me know and I will see that I try the same with the newest anki version.

Furthermore, as you might see I'm doing the selection a bit different than your example in the readme. This example:
```python
selection = col.notes.has_tags(["Japanese", "leech"])
selection = selection.add_tag(["difficult-japanese", "marked"])
col.notes.update(selection)
col.write(modify=True)  # Overwrites your database after creating a backup!
```

results in:
```
----> 2 selection = selection.add_tag(["difficult-japanese", "marked"])

~\Anaconda3\lib\site-packages\pandas\core\generic.py in __getattr__(self, name)
   5137             if self._info_axis._can_hold_identifiers_and_holds_name(name):
   5138                 return self[name]
-> 5139             return object.__getattribute__(self, name)
   5140 
   5141     def __setattr__(self, name: str, value) -> None:

AttributeError: 'Series' object has no attribute 'add_tag'
```

**Comment by klieret** [comment]
Right, the example definitely is just wrong, thanks for reporting this!

**Comment by klieret** [comment]
Yes, as for the tags, this is something I also spotted yesterday. I have already added a quick warning message when writing out that tags will probably have hickups.

I'm not yet sure how to fix this though. I don't understand the database in this respect yet.

**Comment by exc4l** [comment]
I see. Due to the wording or maybe my misinterpretation, I thought it meant problems regarding syncing with the anki-server and not that anki will be unable to find the tagged cards.

`WARNING: Currently AnkiPandas might not be able to tell Anki to sync its tags. So if you added any new tags, you might have to manually tell Anki to sync everything if you want to see and use them in AnkiDroid.
`

**Comment by klieret** [comment]
True, I also didn't expect that there would also be issues with even finding the cards...  (or that even existing tags can't be found). This is quite bad indeed.

**Comment by klieret** [comment]
Though I have to say that I'm still surprised that this should affect existing tags. Because basically we don't even touch the part of the database which might contain some information about the tags, but really only write out the notes table if this is the one that was modified.

**Comment by klieret** [comment]
I can confirm that this is happening. :/

**Comment by klieret** [comment]
Hi @exc4l 

I'm currently not sure if I can fix this properly, as writing info metadata (tags) and the like to anki doesn't work and I don't know how to fix this.

However, here is a workaround: If you first select "Notes > Clear unused notes" and then "Tools > Check Database", it seems to work for me.

**Comment by klieret** [comment]
Can you confirm if this is working?

**Comment by exc4l** [comment]
Doesn't work for me.
Used the exact code from the mwe.
Started Anki up after execution, card browser only showed the leech tag -> 0 notes when selected
Notes -> clear unused tags -> now marked and leech show up in the browser but no notes if searched for them.
After checking the database it's still the same.

Is your message wrong or do you actually mean clear unused notes instead of tags?

**Comment by klieret** [comment]
No, for me, it worked as I described above. Hmm :/

**Comment by klieret** [comment]
This will probably require more reverse engineering about how to access anki's "tags" SQL table. Unfortunately, it's using something called "unicase" in this table yet which I have yet to somehow include in order to even open the table....

**Comment by klieret** [comment]
Perhaps we should look for someone who's working on ankidroid or similar projects to help out here..

**Comment by exc4l** [comment]
I can confirm that this wokaround is working for anki 2.1.35

Guess something changed with the newer versions and it didn't work on my main due to me using an older anki version.

**Comment by klieret** [comment]
Oh, I'm relieved!   Still, I hope we can figure out how to do this properly in the long run....
